apps/api/src/app/environments-v1/usecases/output-renderers/email-output-renderer.usecase.ts

```diff
@@ -84,319 +107,74 @@ export class EmailOutputRendererUsecase {
     const subject = controlSubject as string;
 
     if (disableOutputSanitization) {
-      return { subject, body: htmlWithBranding };
+      return { subject, body: renderedHtml };
     }
 
-    return { subject: sanitizeHTML(subject), body: sanitizeHTML(htmlWithBranding) };
-  }
-
-  private removeTrailingEmptyLines(node: MailyJSONContent): MailyJSONContent {
-    if (!node.content || node.content.length === 0) return node;
-
-    // Iterate from the end of the content and find the first non-empty node
-    let lastIndex = node.content.length;
-    // eslint-disable-next-line no-plusplus
-    for (let i = node.content.length - 1; i >= 0; i--) {
-      const childNode = node.content[i];
-
-      const isEmptyParagraph =
-        childNode.type === 'paragraph' && !childNode.text && (!childNode.content || childNode.content.length === 0);
-
-      if (!isEmptyParagraph) {
-        lastIndex = i + 1; // Include this node in the result
-        break;
-      }
+    return { subject: sanitizeHTML(subject), body: renderedHtml };
+  }
+
+  private async renderWithLayout({
+    stepContent,
+    environmentId,
+    organizationId,
+    disableOutputSanitization,
+    layoutId,
+    payload,
+  }: {
+    stepContent: string;
+    environmentId: string;
+    organizationId: string;
+    disableOutputSanitization: boolean;
+    layoutId?: string;
+    payload: FullPayloadForRender;
+  }): Promise<string> {
+    let layoutControlsEntity: ControlValuesEntity | null = null;
+    // if the step control values have a layoutId then find layout controls entity
+    if (layoutId) {
+      layoutControlsEntity = await this.controlValuesRepository.findOne({
+        _organizationId: organizationId,
+        _environmentId: environmentId,
+        _layoutId: layoutId,
+        level: ControlValuesLevelEnum.LAYOUT_CONTROLS,
+      });
+    } else if (typeof layoutId === 'undefined') {
```

Comment:
Will it handle here the migration period? where layout is null? So it won't try to add it?